### PR TITLE
functionalize for readability

### PR DESCRIPTION
**Pull request by kcho**
Quick fix for
- added internal functions for readability
- MRI and surveys to work for Prescient (still for Prescient, we are checking if there is at least a single csv file. We will need to update this more later)
- removed `mind_audio` column

### CODE REVIEW

File: subject_files_status_for_dpdash.py [hunk 1]
```diff
@@ -94,13 +110,43 @@ def phoenix_files_status(phoenix_dir, out_dir):
...
+def _is_dir(root:Path, subdir: str, pattern: str) -> int:
+    '''check if there is at least one directory matching the pattern'''
+    return _get_dir_count(root, subdir, pattern) > 0
+
+
+def _is_scansheet(root:Path, subdir: str) -> int:
+    '''check if there is a scan sheet for a datatype (subdir)'''
+    return _get_count(
+            root, subdir, f'{root.name}.*.Run_sheet_{subdir}.csv') > 0
+
```
Comment (tashrifbillah) [inline subject_files_status_for_dpdash.py:149]: Lines 120-149 need to go before `phoenix_files_status()`.

File: subject_files_status_for_dpdash.py [hunk 2]
```diff
@@ -94,13 +110,43 @@ def phoenix_files_status(phoenix_dir, out_dir):
...
+    '''get number of directories that mattch glob pattern'''
+    return len([x for x in list((root / subdir).glob(str)) if x.is_dir()])
+
+
+def _get_file_count(root: Path, subdir: str, pattern: str) -> int:
+    '''get number of files that mattch glob pattern'''
+    return len([x for x in list((root / subdir).glob(str)) if x.is_file()])
+
+
+def _is_file(root: Path, subdir: str, pattern: str) -> int:
```
Comment (tashrifbillah) [inline subject_files_status_for_dpdash.py:135]: `is_file`, `is_dir`, and `is_scansheet` return types should be `bool`

File: subject_files_status_for_dpdash.py [hunk 3]
```diff
@@ -94,13 +110,43 @@ def phoenix_files_status(phoenix_dir, out_dir):
             'weekday': ''
         })
         
-        subject_df_tmp = pd.concat([subject_df_tmp, pd.DataFrame(subject_series_tmp).T], axis=1)
-        
+        subject_df_tmp = pd.concat(
+                [subject_df_tmp, pd.DataFrame(subject_series_tmp).T],
+                axis=1)
```
Comment (tashrifbillah) [inline subject_files_status_for_dpdash.py:115]: I am not a big fan of wrapping lines at 80 characters. What does PEP tell us? Couldn't 114 be indented left by one tab to accommodate 115?

File: subject_files_status_for_dpdash.py [hunk 4]
```diff
@@ -45,19 +43,36 @@ def get_summary_from_phoenix(phoenix_dir: Path) -> pd.DataFrame:
...
-    df['mri_ss'] = df.p.apply(lambda x: (x / 'mri' / f'{x.name}.Pronet.Run_sheet_mri.csv').is_file())
-    df['interviews'] = df.p.apply(lambda x: len([x for x in (x / 'interviews').glob('*') if x.is_dir()]) > 0)
-    df['interviews_ss'] = df.p.apply(lambda x: (x / 'interviews' / f'{x.name}.Pronet.Run_sheet_interviews.csv').is_file())
-    df['mind_phone'] = df.p.apply(lambda x: len( [x for x in (x / 'phone').glob('*_activity_*json')] ))
-    df['mind_audio'] = df.p.apply(lambda x: len( [x for x in (x / 'phone').glob('*_activity_*_sound_*.mp3')] ))
-    df['mind_sensor'] = df.p.apply(lambda x: len( [x for x in (x / 'phone').glob('*_sensor_*json')] ))
+    # surveys
+    df['surveys'] = df.p.apply(
+        lambda x: _is_file(x, 'surveys', '*Pronet.json') or 
+                  _is_file(x, 'surveys', '*.csv'))
```
Comment (tashrifbillah) [inline subject_files_status_for_dpdash.py:49]: #### Note for Tashrif for DPdash

If `df['surveys']` is tracked like this, in the `ProNET` configuration card, we have to label it `UPENN Forms Raw Data` and in the `PRESCIENT` configuration card, we have to call it `RPMS Forms Raw Data`. But on the bright side, `*json` and `*csv` files wouldn't occur for the same subject.

**Review by tashrifbillah — CHANGES_REQUESTED** [review]
Tashrif requested a few minor changes

**Comment by tashrifbillah** [comment]
Also Kevin, please check any point we are solving with this PR:
https://github.com/AMP-SCZ/utility/issues/6#issue-1135970573

File: subject_files_status_for_dpdash.py (hunk 3 above)
Comment (kcho) [reply subject_files_status_for_dpdash.py]: [4c6ba88](https://github.com/AMP-SCZ/utility/pull/7/commits/4c6ba88a614861cb1b420767868f0a5b26b4be26)

File: subject_files_status_for_dpdash.py (hunk 2 above)
Comment (kcho) [reply subject_files_status_for_dpdash.py]: [fd8efc7](https://github.com/AMP-SCZ/utility/pull/7/commits/fd8efc79b7410ce999f931a5c43b4a3accabc9e6)

**Comment by tashrifbillah** [comment]
Okay great! Now comes the testing part. Did you already generate the CSVs?

**Comment by kcho** [comment]
> Okay great! Now comes the testing part. Did you already generate the CSVs?

No, not yet. Could you demo testing this at the 11:30AM meeting slot please?

**Comment by tashrifbillah** [comment]
Open issue #9

**Comment by tashrifbillah** [comment]
By PR #10 and this PR, the dashboard got a new and more organized look:

image

**Comment by tashrifbillah** [comment]
The proposed functions made this script more organized. Thank you.

**Pull request merged**